spinalcordtoolbox/reports/qc2.py

```diff
@@ -334,15 +337,47 @@ def sct_deepseg_axial(
     ) if fname_seg_lesion else None
     if fname_seg_lesion:
         img_seg_lesion.data = (img_seg_lesion.data > 0.5) * 1
+    img_qc_seg = resample_nib(
+        image=img_qc_seg,
+        image_dest=img_input,
+        interpolation='linear',
+    ) if fname_qc_seg else None
+
+    # Using -qc-seg mask if available, we remove slices which are empty (except for the first 3 and last 3 slices just around the segmented slices)
+    if fname_qc_seg:
+        # We dilate the mask to keep the first and last 3 slices
+        first_slice = min(np.where(img_qc_seg.data)[0]) - 3
+        last_slice = max(np.where(img_qc_seg.data)[0]) + 3
+        # Now we remove the slices from the input image, the spinal cord mask and the lesion mask
+        img_input.data = img_input.data[first_slice:last_slice]
+        img_seg_sc.data = img_seg_sc.data[first_slice:last_slice]
+        if img_seg_lesion:
+            img_seg_lesion.data = img_seg_lesion.data[first_slice:last_slice]
+        img_qc_seg.data = img_qc_seg.data[first_slice:last_slice]
 
     # Each slice is centered on the segmentation
     logger.info('Find the center of each slice')
-    centers = np.array([center_of_mass(slice) for slice in img_seg_sc.data])
+    # Use the -qc-seg mask if available, otherwise use the spinal cord mask
+    if fname_qc_seg:
+        centers = np.array([center_of_mass(slice) for slice in img_qc_seg.data])
+    else:
+        centers = np.array([center_of_mass(slice) for slice in img_seg_sc.data])
     inf_nan_fill(centers[:, 0])
     inf_nan_fill(centers[:, 1])
 
+    # If -qc-seg is available, use it to generate the radius, otherwise, it is set to the standard value of (15,15)
+    # The radius is chosen as the maximum between (15,15) and the maximum of width and height of the qc-seg mask
+    if fname_qc_seg:
+        widths = [np.max(np.where(slice)[0]) - np.min(np.where(slice)[0]) if np.sum(slice) > 0 else 0 for slice in img_qc_seg.data]
+        heights = [np.max(np.where(slice)[1]) - np.min(np.where(slice)[1]) if np.sum(slice) > 0 else 0 for slice in img_qc_seg.data]
+        widths = [w//2 + 0.1*w//1 for w in widths]
+        heights = [h//2 + 0.1*h//1 for h in heights]
+        radius = (max(15, max(widths)), max(15, max(heights)))
+    else:
+        radius = (15, 15)
```

Comment:
Likewise, the radius code is also duplicated, so it might be good to put that into a function as well:

```suggestion
    # If -qc-seg is available, use it to generate the radius
    radius = get_max_radius(img_qc_seg) if fname_qc_seg else (15, 15)
```

```python
def get_max_radius(img, default=15):
    """Determine the maximum slicewise width/height of the nonzero voxels in img."""
    widths = [np.max(np.where(slice)[0]) - np.min(np.where(slice)[0]) if np.sum(slice) > 0 else 0 for slice in img.data]
    heights = [np.max(np.where(slice)[1]) - np.min(np.where(slice)[1]) if np.sum(slice) > 0 else 0 for slice in img.data]
    widths = [w//2 + 0.1*w//1 for w in widths]
    heights = [h//2 + 0.1*h//1 for h in heights]
    return (max(default, max(widths)), max(default, max(heights)))
```

Reply:
Whatever works best in your opinion !